dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentVersionResource.java

```diff
@@ -73,53 +83,57 @@ public ContentVersionResource() {
     @NoCache
     @Path("/")
     @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
-    public Response findVersions(@Context final HttpServletRequest request,
+    public Response findVersions(@Context final HttpServletRequest request, @QueryParam("inodes") final String inodes,
             @QueryParam("identifier") final String identifier, @QueryParam("groupByLang")final String groupByLangParam, @QueryParam("limit") final int limit)
             throws DotDataException, DotStateException, DotSecurityException {
 
         final boolean groupByLang = "1".equals(groupByLangParam) || BooleanUtils.toBoolean(groupByLangParam);
         final int showing = limit > MAX ? MAX : limit < MIN ? MIN : limit;
         final InitDataObject auth = webResource.init(true, request, true);
-
+        final boolean respectFrontendRoles = PageMode.get(request).respectAnonPerms;
         final User user = auth.getUser();
         try {
 
-            Logger.debug(this,
-                    "Getting versions for identifier: " + identifier + " grouping by language: '" + BooleanUtils.toStringYesNo(groupByLang)+ "' and limit: "+limit);
-
-            final ShortyId shorty = APILocator
-                    .getShortyAPI().getShorty(identifier)
-                    .orElseThrow(() -> new DoesNotExistException(
-                            getFormattedMessage(user.getLocale(), FIND_BY_ID_ERROR_MESSAGE_KEY,
-                                    identifier)));
-            final Identifier identifierObj =
-                    (shorty.type == ShortType.IDENTIFIER) ? APILocator.getIdentifierAPI()
-                            .find(shorty.longId)
-                            : APILocator.getIdentifierAPI().findFromInode(shorty.longId);
+           final Identifier identifierObj = getIdentifier(identifier, user);
 
            ResponseEntityView responseEntityView = null;
-           if(groupByLang){
-               final Map<String, List<Map<String, Object>>> versionsByLang = new HashMap<>();
-               final Map<Long, List<Contentlet>> contentByLangMap = contentletAPI
-                       .findAllVersions(identifierObj, user, false).stream()
-                       .limit(showing)
-                       .collect(Collectors.groupingBy(Contentlet::getLanguageId));
-               contentByLangMap.forEach((langId, contentlets) -> {
-                   final Language lang = languageAPI.getLanguage(langId);
-                   final List<Map<String, Object>> asMaps = contentlets.stream()
-                           .map(this::contentletToMap).collect(Collectors.toList());
-                   versionsByLang.put(lang.toString(), asMaps);
-               });
-
-               responseEntityView = new ResponseEntityView(ImmutableMap.of("versions", versionsByLang));
+           if(null != identifier){
+
+               Logger.debug(this,
```

Comment:
use the lamdba version ()-> for lazy messages

Reply:
done